x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/conversations/append_conversation_messages.ts

```
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+
+import { ConversationResponse, Message } from '@kbn/elastic-assistant-common';
+import { getConversation } from './get_conversation';
+
+export interface AppendConversationMessagesParams {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  conversationIndex: string;
+  existingConversation: ConversationResponse;
+  messages: Message[];
+}
+
+export const appendConversationMessages = async ({
+  esClient,
+  logger,
+  conversationIndex,
+  existingConversation,
+  messages,
+}: AppendConversationMessagesParams): Promise<ConversationResponse | null> => {
```

Consider always throwing instead of returning `null` for some error cases